src/firstboot/firstboot.c

```diff
@@ -1695,6 +1707,8 @@ static int run(int argc, char *argv[]) {
         /* We check these conditions here instead of in parse_argv() so that we can take the root directory
          * into account. */
 
+        if (arg_keymap && !keymap_is_ok(rfd, arg_keymap))
```

Comment:
with the above you'd call it like this:

```
if (arg_keymap && !determine_keymap_validity_func(rfd)(arg_keymap))
        …
```
you get the idea.

(untested, but I think this should just work)